src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -480,21 +507,37 @@ private string CreateTokenPrivate(JObject payload, SigningCredentials signingCre
                 header.Merge(JObject.FromObject(additionalHeaderClaims));
 
             var rawHeader = Base64UrlEncoder.Encode(Encoding.UTF8.GetBytes(header.ToString(Formatting.None)));
-
-            if (SetDefaultTimesOnTokenCreation)
+            try
             {
-                var now = EpochTime.GetIntDate(DateTime.UtcNow);
-                if (!payload.TryGetValue(JwtRegisteredClaimNames.Exp, out _))
-                    payload.Add(JwtRegisteredClaimNames.Exp, now + TokenLifetimeInMinutes * 60);
+                var jsonPayload = JObject.Parse(payload);
```

Comment:
If SetDefaultTimesOnTokenCreation is false, we don't have to parse the payload.

Reply:
Fixed it.